variants/frontend-stimulus/hello_controller.ts

```diff
@@ -0,0 +1,12 @@
+import { Controller } from '@hotwired/stimulus';
+
+// To see this controller in action, add the following to the DOM:
+//
+//   <div data-controller="hello"></div>
+
+// eslint-disable-next-line import/no-anonymous-default-export
+export default class extends Controller {
+  connect() {
+    this.element.textContent = 'Hello World!';
+  }
+}
```

Comment:
@G-Rath Currently this file is identical to the JS version. Are there changes I should make to set up a dev for success with a TS controller?

Reply:
We probably want to include `GenericController`:

```
import { Controller } from 'stimulus';

export default class GenericController<
  TElement extends Element = Element
> extends Controller {
  public override get element(): TElement {
    return super.element as TElement;
  }
}
```

Because Stimulus currently defines the type of `element` as just `Element` so you can't properly use properties that are only on some subsets without casting.

(I believe there's an issue open over on the stimulus repo to make it generic, and I'm planning to push for that once we've required it on a few projects to make it easy to show it is useful - @mischa-s and @joshmcarthur can better speak to how important it is actually is)